File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
         for(int i = 0; i < haystack.length(); i++){
            if(haystack.charAt(i) == needle.charAt(0) && needle.length()<=haystack.length()-i){
                int temp = i, index = 0;
                while(index<needle.length()){
                    if(needle.charAt(index) != haystack.charAt(temp++))break;
                    index++;
                    if(index == needle.length()) return i;
                }
            }
        }
        return -1;
    }
}

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {

        int tl=haystack.size();  //Text length
        int pl=needle.size();   //Pattern Length
        int ans=-1;
        if(pl==0) return 0;
        for(int i=0;i<=tl-pl;i++){
            int j;
            for(j=0;j<pl;j++){
                if(haystack[i+j]!=needle[j]) break;    
            }
            if(j==pl) {
                ans=i;
                break;
            }
        }
        return ans;    
    }
};